c/dataservice.c

```diff
@@ -214,18 +214,25 @@ WebPlugin *makeWebPlugin(char *pluginLocation, JsonObject *pluginDefintion, Inte
     for (int i = 0; i < jsonArrayGetCount(dataServices); i ++) {
       JsonObject *serviceDef = jsonArrayGetObject(dataServices, i);
       char *type = jsonObjectGetString(serviceDef, "type");
-
-      if (!type || !strcmp(type, "service")) {
-        plugin->dataServiceCount ++;
-      } else if (!strcmp(type, "group")) {
-        JsonArray* group = jsonObjectGetArray(serviceDef, "subservices");
-        if (group) {
-          plugin->dataServiceCount += jsonArrayGetCount(group);
-        }
-      } else if (!strcmp(type,"nodeService")){
-        /* Node services will be handled by node without ever going to the MVD server */
+      char *serviceName = jsonObjectGetString(serviceDef, "name");
+      if (!serviceName) {
+        // Return a null plugin since the dataservice definition is not correct.
+        printf("*** PANIC: Returning NULL for plugin. Check pluginDefinition for correct dataservice syntax ***\n");
+        plugin = NULL;
+        return plugin;
       } else {
-        /* import, ignore */
+        if (!type || !strcmp(type, "service")) {
```

Comment:
I think this area of the code could use a minor improvement:
if !type: issue a warning, return NULL
if group: as-is, not sure how this is used personally.
if import || router || nodeService, ignore
else print warning saying type unknown, but don't return NULL... since it may just be something the server can't handle but the nodejs server can instead.

Reply:
This should be all set. Thanks.